### ISSUE TITLE
dice coefficient not improving accuracy, stuck at 50.1%

### ISSUE THREAD

**Issue by I-CANT-CODE** [jakeret/tf_unet#250]
Wondering why the dice coefficient isn't working. it pretty much stays stuck at 50.1% accuracy and -.4986 = loss each epoch. prediction images aren't doing well either. when I keep everything else the same except for the cost function, the network trains up to 80% accuracy using cross entropy. any ideas?

**Comment by jis478** [comment]
Is your data imbalanced? Then can you try a modified version I've made based on this repo? I'm wondering my approach also works for other imbalanced problems.

You can simply try the below jupyter notebook file.
https://github.com/jis478/Tensorflow/tree/master/Unet_modified/example/membrane/code/Unet_modified_execution.ipynb

**Comment by I-CANT-CODE** [comment]
@jis478 Hey, so I was looking at your repo, but I can't find the portion of the code where you modify the dice cost. Can you link me to specifically where it Is modified?

**Comment by I-CANT-CODE** [comment]
@jis478 Hey, so I looked at your code and found where you add a modified dice coefficient. I added this to my version of the code and I'm running it, but it doesn't seem to be helping.

**Comment by I-CANT-CODE** [comment]
didn't mean to close the issue my bad

**Comment by jis478** [comment]
Sorry for my late reply. You can have a look at https://github.com/jis478/Tensorflow/blob/master/Unet_modified/example/membrane/code/modeling.py

The jupyter notebook I've provided before is a result of the improved version. You can first try with the jupyter notebook to see the improvements.

**Comment by stiv-yakovenko** [comment]
Dear jis478, thank you so much for contributing your loss to the community, it helped me to solve my segmentation problem, otherwise it didn't work. did you have idea to merge this to this repository directly?

**Comment by jis478** [comment]
You can give my repo a star then :) I will make a pull request to the original unet code as soon as possible. Thanks!
